unit-tests/game-api-tests.js

```diff
@@ -1,32 +1,23 @@
-//describe('gameApiTest', function(){
-//
-//    var gameApi;
-//
-//    beforeEach(function(){
-//        module('tombola.noughtAndCrosses.services');
-//        inject(function($injector){
-//            gameApi = $injector.get('gameApi');
-//        });
-//    });
-//
-//    it('Should contain gameApi service', function(){
-//        expect(noughtsAndCrossesApp.gameApi)
-//            .to.equal('noughtsAndCrossesApp.service(gameApi,function ($http,$q,gameApiConstants)')
-//    });
-//
-//    it('makeMove function contains appropriate playerNumber and squareNumber', function(){
-//        expect(noughtsAndCrossesApp.gameApi.makeMove)
-//            .to.equal(gameApi.makeMove.playerNumber = 'playerNumber',gameApi.makeMove.squareNumber = 'squareNumber');
-//    });
-//
-//    it('newGame function contains constants', function(){
-//        expect(noughtsAndCrossesApp.gameApi.newGame)
-//            .to.equal(gameApi.newGame.player1 = 'player1Type',gameApi.newGame.player2 = 'player2Type');
-//    });
-//
-//    it('gameApi service contains promise', function(){
-//        expect(noughtsAndCrossesApp.gameApi)
-//            .to.equal(gameApi.callServer);
-//    });
-//
-//});
+describe('gameApiTest', function(){
+
+    var gameApi;
+    var gameModel;
+
+    beforeEach(function(){
+        module('tombola.noughtAndCrosses.services');
+        inject(function($injector){
+            gameApi = $injector.get('gameApi');
+            gameModel = $injector.get('NoughtsAndCrosses/unit-tests/mocks/mock-game-model.js')
+        });
+    });
+
+    it('newGame function should reset the game board', function(){
+        gameApi.newGame();
+        expect(gameModel.gameboard).to.equal('000000000');
+    });
+
+    it('makeMove function finds currentPlayer and squareNumber placement',function(){
+        gameApi.makeMove();
+        expect(gameModel.currentPlayer).to.equal(1||2);
```

Comment:
Shit or get off the pot :) ! You either want (current player ===1) or  (current player ===2) not (current player ===1 || current player ===2) - You should know which.
Again the we're testing the API though it's effects rather than what it does. We should really just make a move and make sure the expected response object comes back on the mock.